src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/StringParsingHelpers.Statistics.cs

```diff
@@ -413,27 +413,29 @@ internal static IPInterfaceStatisticsTable ParseInterfaceStatisticsTableFromFile
                     string line = sr.ReadLine()!;
                     if (line.Contains(name))
                     {
-                        string[] pieces = line.Split(new char[] { ' ', ':' }, StringSplitOptions.RemoveEmptyEntries);
+                        Span<Range> pieces = stackalloc Range[18]; // [0] skipped, [1]-[16] used, +1 to ensure any additional segment goes into [17]
+                        ReadOnlySpan<char> lineSpan = line;
+                        pieces = pieces.Slice(0, lineSpan.SplitAny(pieces, " :", StringSplitOptions.RemoveEmptyEntries));
 
                         return new IPInterfaceStatisticsTable()
                         {
```

Comment:
is worth the effort here to first parse `lineSpan[pieces[16]` to avoid bound checks for the rest?

Reply:
I don't think so.

Reply:
codegen of SplitAny Span<Range&gt; is 5x worse https://godbolt.org/z/5essMKz64 (build # 6a1ba1c305f81c535b25cbed6725c31e761f2c24). if godbolt isn't lying then this is big enough difference to keep using string.Split maybe just switch to its `count` overload?

Reply:
You're comparing apples to oranges.  First, the comparison isn't what's being done here, e.g. you're doing `line[pieces[i]]` rather than `lineSpan[pieces[i]]`; the substrings never need to be materialized, but you are in both versions; in this PR, the spans are being parsed directly.  Second, the vast majority of that "5x" is actually code for Range, Span, ReadOnlySpan, etc., has little to do with this particular usage, and will already be included in an app using this... the actual difference in the size of the asm for M in the SplitAny version is ~50 bytes. Third, you can see from this benchmark how much faster and less allocation the new version entails:
```C#
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;
using System;
using System.Globalization;
using System.Runtime.CompilerServices;

[MemoryDiagnoser(false)]
[SkipLocalsInit]
public partial class Program
{
    static void Main(string[] args) => BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);

    private string _value = "0:1:2:3:4:5:6:7:8:9:10:11:12:13:14:15:16:17";

    [Benchmark(Baseline = true)]
    public long StringSplit()
    {
        string[] strings = _value.Split(new char[] { ' ', ':' });
        return
            Parse(strings[1]) + Parse(strings[2]) + Parse(strings[3]) + Parse(strings[4]) +
            Parse(strings[5]) + Parse(strings[6]) + Parse(strings[7]) + Parse(strings[8]) +
            Parse(strings[9]) + Parse(strings[10]) + Parse(strings[11]) + Parse(strings[12]) +
            Parse(strings[13]) + Parse(strings[14]) + Parse(strings[15]) + Parse(strings[16]);
    }

    [Benchmark]
    public long SpanSplit()
    {
        Span<Range> strings = stackalloc Range[18];
        ReadOnlySpan<char> valueSpan = _value;
        valueSpan.SplitAny(strings, " :");
        return
            Parse(valueSpan[strings[1]]) + Parse(valueSpan[strings[2]]) + Parse(valueSpan[strings[3]]) + Parse(valueSpan[strings[4]]) +
            Parse(valueSpan[strings[5]]) + Parse(valueSpan[strings[6]]) + Parse(valueSpan[strings[7]]) + Parse(valueSpan[strings[8]]) +
            Parse(valueSpan[strings[9]]) + Parse(valueSpan[strings[10]]) + Parse(valueSpan[strings[11]]) + Parse(valueSpan[strings[12]]) +
            Parse(valueSpan[strings[13]]) + Parse(valueSpan[strings[14]]) + Parse(valueSpan[strings[15]]) + Parse(valueSpan[strings[16]]);
    }

    [Benchmark]
    public long SpanSplitReorder()
    {
        Span<Range> strings = stackalloc Range[18];
        ReadOnlySpan<char> valueSpan = _value;
        valueSpan.SplitAny(strings, " :");
        return
            Parse(valueSpan[strings[16]]) +
            Parse(valueSpan[strings[1]]) + Parse(valueSpan[strings[2]]) + Parse(valueSpan[strings[3]]) + Parse(valueSpan[strings[4]]) +
            Parse(valueSpan[strings[5]]) + Parse(valueSpan[strings[6]]) + Parse(valueSpan[strings[7]]) + Parse(valueSpan[strings[8]]) +
            Parse(valueSpan[strings[9]]) + Parse(valueSpan[strings[10]]) + Parse(valueSpan[strings[11]]) + Parse(valueSpan[strings[12]]) +
            Parse(valueSpan[strings[13]]) + Parse(valueSpan[strings[14]]) + Parse(valueSpan[strings[15]]);
    }

    private static long Parse(string value) =>
        (long)Math.Min(long.MaxValue, ulong.Parse(value, CultureInfo.InvariantCulture));

    private static long Parse(ReadOnlySpan<char> value) =>
        (long)Math.Min(long.MaxValue, ulong.Parse(value, CultureInfo.InvariantCulture));
}
```

|           Method |     Mean |   Error |  StdDev | Ratio | Allocated | Alloc Ratio |
|----------------- |---------:|--------:|--------:|------:|----------:|------------:|
|      StringSplit | 427.9 ns | 4.24 ns | 3.76 ns |  1.00 |     696 B |        1.00 |
|        SpanSplit | 246.7 ns | 3.03 ns | 2.68 ns |  0.58 |         - |        0.00 |
| SpanSplitReorder | 246.6 ns | 3.24 ns | 2.87 ns |  0.58 |         - |        0.00 |

And that the impact of moving up the last indexing to avoid the bounds check isn't particularly meaningful here given how much other work is being done.